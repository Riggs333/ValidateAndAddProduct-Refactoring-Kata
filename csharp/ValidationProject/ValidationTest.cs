using System;
using System.Collections.Generic;
using System.Text;
using Xunit;
using ApprovalTests;
using ApprovalTests.Reporters;

namespace Validation
{
    [UseReporter(typeof(DiffReporter))]
    public class ValidationTest
    {
        [Fact]
        void ValidateAndAdd() {
            // Arrange
            var productData = new ProductFormData("Sample product",
                "Lipstick", 5D, 10D, false);
            var db = new FakeDatabase();
            var sut = new ProductService(db);

            // Act
            var response = sut.ValidateAndAdd(productData);

            // Assert
            var productString = "";
            if (db.Product != null) {
                productString = db.Product.ToString();
            }
            var responseAndProduct = response + " " + productString;
            Approvals.Verify(responseAndProduct);

        }

    }
}
