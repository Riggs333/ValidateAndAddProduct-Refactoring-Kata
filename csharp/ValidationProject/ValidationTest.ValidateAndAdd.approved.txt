Response{productId=0, statusCode=0, message='Product Successfully Added'} Product{name='Sample product', type='Lipstick', weight=5, family=LIPS, range=BUDGET}